src/test/java/com/spotify/dns/NoCachingLookupFactoryTest.java

```diff
@@ -0,0 +1,54 @@
+package com.spotify.dns;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.xbill.DNS.*;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class NoCachingLookupFactoryTest {
```

Comment:
This test class doesn't validate the key behaviour which I understand to be disabling the DNS cache. I expect that's pretty hard to test in practice. Did you try?